### PR TITLE
Update the existing NETSDK 1045 error message to link to the download page

### LINKED ISSUE #22959: Enhance the call to action of the UnsupportedTargetFrameworkVersion error for users
### Is your feature request related to a problem? Please describe.

See [this twitter thread from Eric Sink](https://twitter.com/ericlaw/status/1468571781862674438), but the gist is:
* user has 6.0.100 installed
* user has a project targeting net6.0
* user opens the project in VS 2019
* user is presented with

> NETSDK1045: The current .NET SDK does not support targeting .NET 6.0.  Either target .NET 5.0 or lower, or use a version of the .NET SDK that supports .NET 6.0.

### Describe the solution you'd like
I'd like a more direct call to action in this message, customized by the context of the project that's being built.

[Here](https://github.com/dotnet/sdk/blob/4cac4e4cd5a04bfd485821b06ea2d3f218c32cfe/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L140-L148) is the current target that fires this SDK error.

For the general case of CLI users, since we know the desired SDK (in this case 6.0), we could add a direct link to `https://dotnet.microsoft.com/download/download/dotnet/6.0` (the SDK version is at the end here, we would template this as part of the message) to navigate the user direct to where the solution for their problem is likely to be.

For the specific case of VS users, we could have a slightly different treatment. First, we should add a check for `BuildingInsideVisualStudio == false` to the set of conditions on the current target.  Then, we should have a sibling target specifically for the VS use case. When running under VS, we have access to the `BuildingInsideVisualStudio` and `VisualStudioVersion` properties, as shown in [other targets](https://github.com/dotnet/sdk/blob/6a7e4caafea6d683ac76d3aec0fe53432df6ed79/src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.Compilation.targets#L93). I propose that we have a map of VS Version to minimum required VS version, and in this new VS-specific target, if the current VS version doesn't support the SDK being targeted, be explicit about that in the message and direct the user to download the required version. Getting a direct link for this seems harder, as most VS acquisition seems focused on the installer app.

### PR DESCRIPTION
**Pull request by marcpopMSFT**
## Description
Fixes #22959

We want to improve the experience for customers who are accidentally using 17.7 to try to target .NET 8. Currently the error says the customer should use an SDK that supports .NET 8 but we want to be more specific to tell customers to use 17.8 or newer.

## Customer Impact

Current error experience (which will still be seen when using .net 7 in 17.8 or when building from the command line with 7.0.4xx)
   _<value>NETSDK1045: The current .NET SDK does not support targeting .NET 8.  Either target .NET 7 or lower, or use a version of the .NET SDK that supports .NET 8.</value>_

New experience:
   _<value>NETSDK1209: The current Visual Studio version does not support targeting .NET 8.  Either target .NET 7 or lower, or use Visual Studio version 17.8 or higher</value>_
 

## Regression?
* [ ] Yes
* [x] No

## Risk
* [ ] High
* [ ] Medium
* [x] Low

## Verification

* [x] manual
* [ ] automated tests

Can't automate this as the lab will update to 17.8 as soon as it's available and any test we had would start failing.

---
This is an odd one as the request in #22959 is to provide a better error when targeting a newer runtime than the current VS (and SDK) support. That means having an older SDK know about a newer one. I don't think we can generalize this as the minimum VS version will keep changing.

Detect we're building in VS and that the target framework is too high and the VS is too low and provide an error telling customers to update their Visual Studio Version.

CC @baronfel on how he feels about this experience.

### CODE REVIEW

**Comment by marcpopMSFT** [comment]
@baronfel any thoughts on this error improvement?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props [hunk 1]
```diff
@@ -28,6 +28,11 @@ Copyright (c) .NET Foundation. All rights reserved.
                                             Condition="$([MSBuild]::VersionGreaterThanOrEquals($(MSBuildVersion), '17.4.0'))"/>
     </ItemGroup>
 
+    <PropertyGroup>
+        <UnsupportedTargetFrameworkVersion>8.0</UnsupportedTargetFrameworkVersion>
+        <MinimumVisualStudioVersionForUnsupportedTargetFrameworkVersion>17.8</MinimumVisualStudioVersionForUnsupportedTargetFrameworkVersion>
```
Comment (baronfel) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props:33]: Is there a way we can know when these need bumping automatically? Or do we just add it to the 'things we bump when we version' list?

**Review by baronfel — APPROVED** [review]

**Comment by baronfel** [comment]
Love it, thanks for knocking this out @marcpopMSFT.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props (hunk 1 above)
Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props:33]: I don't think there's a way to know for the VS version as that version can change with each major. I think we can expect 4 releases a year most of the time so it's probably predictable for a couple of years but would have to be updated with each major VS release. Might be safer to just update every year.

For the UnsupportedTFM, I might be able to calculate it from the NETCoreAppMaximumVersion value. Is it worth doing if we have to update the VS version each time?

Comment (baronfel) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props:33]: No, you make a good case for keeping it manual. I wasn't sure if we had the same data available elsewhere easily or not, that's all.

File: src/Tasks/Common/Resources/Strings.resx [hunk 2]
```diff
@@ -280,7 +280,7 @@
     <comment>{StrBegin="NETSDK1044: "}</comment>
   </data>
   <data name="UnsupportedTargetFrameworkVersion" xml:space="preserve">
-    <value>NETSDK1045: The current .NET SDK does not support targeting {0} {1}.  Either target {0} {2} or lower, or use a version of the .NET SDK that supports {0} {1}.</value>
+    <value>NETSDK1045: The current .NET SDK does not support targeting {0} {1}.  Either target {0} {2} or lower, or use a version of the .NET SDK that supports {0} {1}. Download the .NET SDK from https://aka.ms/download-dotnet</value>
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply src/Tasks/Common/Resources/Strings.resx:283]: I do want to appreciate that we didn't put a period at the end of the URL

[the remark this answers is not in the document]
Comment (richlander) [reply src/Tasks/Common/Resources/Strings.resx:283]: I didn't notice that, but it is a great point.

[the remark this answers is not in the document]
Comment (marcpopMSFT) [reply src/Tasks/Common/Resources/Strings.resx]: Updated to use dotnet/download instead (with no period)

**Review by baronfel — APPROVED** [review]

**Pull request merged**